community-connectors/github/CustomCache.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * Custom cache backed by Google Sheets.
+ *
+ * @param {string} organization - The organization name.
+ * @param {string} repository - The repository name.
+ * @return {object} the cache object.
+ */
+function CustomCache(organization, repository) {
+  var spreadsheet;
+  var userProperties = CustomCache.getPropertiesServiceNS().getUserProperties();
+  var scriptId = userProperties.getProperty(CustomCache.SCRIPT_ID_KEY);
+  if (scriptId === null) {
+    spreadsheet = CustomCache.getSpreadsheetAppNS().create(CustomCache.SPREADSHEET_NAME);
+    userProperties.setProperty(CustomCache.SCRIPT_ID_KEY, spreadsheet.getId());
+  } else {
+    try {
+      spreadsheet = CustomCache.getSpreadsheetAppNS().openById(scriptId);
+    } catch (e) {
+      if (e.message.match(/is missing/)) {
+        spreadsheet = CustomCache.getSpreadsheetAppNS().create(CustomCache.SPREADSHEET_NAME);
+        userProperties.setProperty(CustomCache.SCRIPT_ID_KEY, spreadsheet.getId());
+      } else {
+        console.log(e);
+        throw Error('The sheet could not be created.');
+      }
+    }
+  }
+  this.spreadsheet = spreadsheet;
+  var sheetName = CustomCache.buildSheetName(organization, repository);
+  var sheet = spreadsheet.getSheetByName(sheetName);
+  if (sheet === null) {
+    spreadsheet.insertSheet(sheetName);
+  }
+  this.sheet = spreadsheet.getSheetByName(sheetName);
+  return this;
+}
+
+CustomCache.SCRIPT_ID_KEY = 'sheets cache script id';
+
+CustomCache.SPREADSHEET_NAME = 'user cache for GitHub.com connector.';
```

Comment:
If the CustomCache.js is reusable, should this be a more generic name or defined in the main module?

Reply:
Right now, the CustomCache isn't quite ready to be reusable. Until I know which parts make sense to keep between different connectors, I'm planning to punt on things like this.